pkg/controller/data_upload_controller.go

```diff
@@ -292,19 +292,20 @@ func (r *DataUploadReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	} else if du.Status.Phase == velerov2alpha1api.DataUploadPhaseInProgress {
 		log.Info("Data upload is in progress")
 		if du.Spec.Cancel {
-			fsBackup := r.dataPathMgr.GetAsyncBR(du.Name)
-			if fsBackup == nil {
-				return ctrl.Result{}, nil
-			}
 			log.Info("Data upload is being canceled")
 
-			// Update status to Canceling.
+			// Update status to Canceled
 			original := du.DeepCopy()
-			du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceling
```

Comment:
There are two cases:
1. If the dataPath is still there (`fsBackup` is not nil), we cannot update the DUCR to cancelled directly. Instead, we need to notify the dataPath (by calling `fsBackup.Cancel()`), wait for it to cancel the data transfer and then it will call back `OnDataUploadCancelled`
2. If the dataPath is not there, we can cancel the DUCR in the main flow

Both of the cases must be supported. However:

- For the existing code, case 2 is not supported
- For the changes in this PR, case 1 will not be supported

So please reconstruct the code to support the both case